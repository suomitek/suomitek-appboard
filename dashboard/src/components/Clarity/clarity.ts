// Code from: https://stackblitz.com/edit/react-ts-wrapper-5w8nhf
import { CdsButton as Button } from "@clr/core/button";
import { CdsIcon as Icon, ClarityIcons as ClrIcons } from "@clr/core/icon-shapes";
import {
  angleIcon,
  applicationsIcon,
  backupRestoreIcon,
  bundleIcon,
  checkCircleIcon,
  circleArrowIcon,
  ClarityIcons as Icons,
  clusterIcon,
  copyToClipboardIcon,
  deployIcon,
  exclamationCircleIcon,
  exclamationTriangleIcon,
  eyeHideIcon,
  eyeIcon,
  fileGroupIcon,
  fileIcon,
  heartIcon,
  helpIcon,
  infoCircleIcon,
  plusCircleIcon,
  refreshIcon,
  rewindIcon,
  searchIcon,
  timesCircleIcon,
  timesIcon,
  trashIcon,
  uploadCloudIcon,
} from "@clr/core/icon-shapes";

import "@clr/core/button";
import "@clr/core/icon";
import { createReactComponent } from "./converter/reactWrapper";

type CdsIconType = Icon;
export const CdsIcon = createReactComponent<CdsIconType>("cds-icon");
export const ClarityIcons = ClrIcons;

type CdsButtonType = Button & HTMLButtonElement;
export const CdsButton = createReactComponent<CdsButtonType>("cds-button");

Icons.addIcons(
  angleIcon,
  bundleIcon,
  checkCircleIcon,
  circleArrowIcon,
  clusterIcon,
  exclamationTriangleIcon,
  eyeHideIcon,
  eyeIcon,
  fileGroupIcon,
  helpIcon,
  infoCircleIcon,
  rewindIcon,
  searchIcon,
  trashIcon,
  uploadCloudIcon,
  applicationsIcon,
  heartIcon,
  exclamationCircleIcon,
  timesIcon,
  deployIcon,
  fileIcon,
  copyToClipboardIcon,
  backupRestoreIcon,
  refreshIcon,
  plusCircleIcon,
  timesCircleIcon,
);
