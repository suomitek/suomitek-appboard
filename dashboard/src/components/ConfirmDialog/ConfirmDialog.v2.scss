.confirmation-modal {
  text-align: center;
  overflow-y: hidden;
  span {
    color: var(--clr-p-color, black);
  }
  &-buttons {
    margin-top: 0.6rem;
    cds-button {
      margin: 0 0.6rem 0 0.6rem;
    }
  }
}
