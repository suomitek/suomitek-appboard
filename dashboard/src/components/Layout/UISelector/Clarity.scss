.color-icon-info {
  cds-icon {
    --color: var(--clr-alert-info-icon-color, #0072a3);
  }
}

.color-icon-success {
  cds-icon {
    --color: var(--clr-alert-success-icon-color, #3c8500);
  }
}

.color-icon-warning {
  cds-icon {
    --color: var(--clr-alert-warning-icon-color, #666666);
  }
}

.color-icon-danger {
  cds-icon {
    --color: var(--clr-alert-danger-icon-color, #c21d00);
  }
}

.center {
  display: grid;
  height: 100%;
  width: 100%;
  justify-content: center;
  align-items: center;
}
