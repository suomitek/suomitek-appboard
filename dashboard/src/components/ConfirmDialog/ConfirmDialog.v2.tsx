import { CdsButton } from "components/Clarity/clarity";
import Alert from "components/js/Alert";
import Modal from "components/js/Modal/Modal";
import React from "react";
import LoadingWrapper from "../LoadingWrapper/LoadingWrapper.v2";
import "./ConfirmDialog.v2.css";

interface IConfirmDialogProps {
  modalIsOpen: boolean;
  loading: boolean;
  extraElem?: JSX.Element;
  confirmationText: string;
  confirmationButtonText?: string;
  error?: Error;
  onConfirm: () => any;
  closeModal: () => any;
}

function ConfirmDialog({
  modalIsOpen,
  loading,
  extraElem,
  confirmationButtonText,
  confirmationText,
  onConfirm,
  closeModal,
  error,
}: IConfirmDialogProps) {
  return (
    <Modal showModal={modalIsOpen} onModalClose={closeModal}>
      {error && <Alert theme="danger">Found error: {error.message}</Alert>}
      {loading === true ? (
        <div className="confirmation-modal">
          <span>Loading, please wait</span>
          <LoadingWrapper loaded={false} />
        </div>
      ) : (
        <div className="confirmation-modal">
          <span>{confirmationText}</span>
          {extraElem}
          <div className="confirmation-modal-buttons">
            <CdsButton action="outline" type="button" onClick={closeModal}>
              Cancel
            </CdsButton>
            <CdsButton status="danger" type="submit" onClick={onConfirm}>
              {confirmationButtonText || "Delete"}
            </CdsButton>
          </div>
        </div>
      )}
    </Modal>
  );
}

export default ConfirmDialog;
