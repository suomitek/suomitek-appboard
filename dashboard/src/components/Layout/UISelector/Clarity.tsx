import * as React from "react";

import "../../../clr-ui.min.css";
import "./Clarity.css";

const CSSSelector: React.FC = () => null;

export default CSSSelector;
