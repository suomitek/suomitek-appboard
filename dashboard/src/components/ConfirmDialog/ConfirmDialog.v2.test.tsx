import { CdsButton } from "components/Clarity/clarity";
import { mount } from "enzyme";
import context from "jest-plugin-context";
import * as React from "react";
import itBehavesLike from "../../shared/specs";
import ConfirmDialog from "./ConfirmDialog.v2";

const defaultProps = {
  loading: false,
  modalIsOpen: true,
  onConfirm: jest.fn(),
  closeModal: jest.fn(),
};

context("when loading is true", () => {
  itBehavesLike("aLoadingComponent", {
    component: ConfirmDialog,
    props: { ...defaultProps, loading: true },
  });
});

it("should modify the default confirmation text", () => {
  const wrapper = mount(
    <ConfirmDialog {...defaultProps} confirmationText="Sure?" confirmationButtonText="Sure!" />,
  );
  expect(wrapper.find(".confirmation-modal")).toIncludeText("Sure?");
  expect(wrapper.find(CdsButton).filterWhere(d => d.text() === "Sure!")).toExist();
});
