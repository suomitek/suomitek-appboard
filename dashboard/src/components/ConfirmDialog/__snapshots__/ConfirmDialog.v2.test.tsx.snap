// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when loading is true loading spinner matches the snapshot 1`] = `
<Modal
  hideCloseButton={false}
  modalSize="md"
  onModalClose={[MockFunction]}
  showModal={true}
  staticBackdrop={true}
>
  <div
    className="confirmation-modal"
  >
    <span>
      Loading, please wait
    </span>
    <LoadingWrapper
      loaded={false}
    />
  </div>
</Modal>
`;
